### PR TITLE
Adapt to new widgets feature "theme config"

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ✔️
| Fixed issues  | -

### CODE REVIEW

File: config/params.php [hunk 1]
```diff
@@ -4,9 +4,9 @@
 
 return [
     'yiisoft/form' => [
-        'configs' => [
+        'themeConfigs' => [
```
Comment (samdark) [inline config/params.php:7]: I think these names were alright.

File: docs/creating-forms.md [hunk 2]
```diff
@@ -111,12 +111,12 @@ $this->setTitle($title);
             ->csrf($csrf)
             ->id('form-auth-login')
             ->open() ?>
-            <?= Field::text($formModel, 'login')
+            <?= ThemeDispatcher::text($formModel, 'login')
```
Comment (samdark) [inline docs/creating-forms.md:114]: That's does not read as good as it was before the change.

File: tests/ConfigTest.php [hunk 3]
```diff
@@ -48,13 +45,13 @@ public function testCustomParams(): void
...
         ];
-        $params['yiisoft/form']['defaultConfig'] = 'simple';
+        $params['yiisoft/form']['defaultThemeName'] = 'simple';
 
         $bootstrapList = $this->getBootstrapList($params);
         $bootstrap = array_shift($bootstrapList);
         $bootstrap(new SimpleContainer());
 
-        $input = Field::hidden(new HiddenForm(), 'key');
+        $input = Hidden::widget()->formAttribute(new HiddenForm(), 'key');
```
Comment (samdark) [inline tests/ConfigTest.php:54]: That is significant regression is readability.

Comment (vjik) [reply tests/ConfigTest.php:54]: In master `Field` merge package-specific configurations (common configuration in field factory, individual configuration in field factory, configuration at the place of use) and pass it to widget factory from `yiisoft/widget`.

In this PR use new feature of `yiisoft/widget` (PR)In master `Field` merge package-specific configurations (common configuration in field factory, individual configuration in field factory, configuration at the place of use) and pass to `WidgetFactory` from `yiisoft/widget`.

This PR use new feature of `yiisoft/widget` (see https://github.com/yiisoft/widget/pull/84). Now package-specific configuration merged into widget, `Field` not need for it.

But we can create new `Field` that will be used only as shortcutrs. How is it done in Yii HTML (helper `HTML`).

Comment (samdark) [reply tests/ConfigTest.php:54]: Here I'm talking only about how it looks from user perspective. HTML and widgets are used a lot and developer types their code and reads this code constantly so the following is a significant difference:

```php
Field::hidden(new HiddenForm(), 'key');

// vs

Hidden::widget()->formAttribute(new HiddenForm(), 'key');
```

Comment (vjik) [reply tests/ConfigTest.php:54]: Yes, I agree with it.

Comment (vjik) [reply tests/ConfigTest.php:54]: Goal of PR make `Field::hidden(new HiddenForm(), 'key');` === `Hidden::widget()->formAttribute(new HiddenForm(), 'key');`.

In master `Field::...` use package-specific configuration, but `Hidden::...` don't use. In PR both calls returns same result.

Comment (samdark) [reply tests/ConfigTest.php:54]: Then why do we change all tests and phpdoc to use `widget()`?

File: config/params.php (hunk 1 above)
Comment (vjik) [reply config/params.php]: revert

File: docs/creating-forms.md (hunk 2 above)
Comment (vjik) [reply docs/creating-forms.md]: fixed

File: tests/ConfigTest.php (hunk 3 above)
Comment (vjik) [reply tests/ConfigTest.php]: > Then why do we change all tests and phpdoc to use `widget()`?

Not need. I'm added `Field` with static shortcut methods.

File: src/Field.php [hunk 4]
```diff
@@ -33,238 +32,182 @@
 use Yiisoft\Form\Field\Textarea;
 use Yiisoft\Form\Field\Url;
 
-use function array_key_exists;
-
-final class Field
+class Field
```
Comment (samdark) [inline src/Field.php:35]: Is the idea to be able to `extend` it in your app to add more shortcuts?

File: src/ThemeDispatcher.php [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+declare(strict_types=1);
+
+namespace Yiisoft\Form;
+
+use RuntimeException;
+
+use function array_key_exists;
+
+final class ThemeDispatcher
```
Comment (samdark) [inline src/ThemeDispatcher.php:11]: It doesn't dispatch anything so... maybe a theme factory?

File: src/Field.php (hunk 4 above)
Comment (vjik) [reply src/Field.php:35]: Yes. For add shortcuts for non-core fields.

File: src/ThemeDispatcher.php (hunk 5 above)
Comment (vjik) [reply src/ThemeDispatcher.php:11]: `ThemeContainer` ?

Comment (samdark) [reply src/ThemeDispatcher.php:11]: Better than dispatcher, yes.

Comment (vjik) [reply src/ThemeDispatcher.php]: done

**Review by samdark — APPROVED** [review]

**Pull request merged**